package/MDAnalysis/core/groups.py

```diff
@@ -28,48 +43,81 @@ def copy_class(newname, *parents):
         return type(newname, tuple(parents), {})
 
     classdict = {}
+    # The 'GB' middle man is needed so that a single topologyattr
+    #  patching applies automatically to all groups.
     GB = classdict['group'] = copy_class(
         'Group', GroupBase)
     AG = classdict['atomgroup'] = copy_class(
-        'AtomGroup', GB, AtomGroup)
+        'AtomGroupBase', GB)
     RG = classdict['residuegroup'] = copy_class(
-        'ResidueGroup', GB, ResidueGroup)
+        'ResidueGroupBase', GB)
     SG = classdict['segmentgroup'] = copy_class(
-        'SegmentGroup', GB, SegmentGroup)
+        'SegmentGroupBase', GB)
+    # In the current Group-centered topology scheme no attributes apply only
+    #  to ComponentBase, so no need to have a 'CB' middle man.
+    #CB = classdict['component'] = copy_class(
+    #    'ComponentBase', ComponentBase)
     A = classdict['atom'] = copy_class(
-        'Atom', Atom)
+        'AtomBase', ComponentBase)
     R = classdict['residue'] = copy_class(
-        'Residue', Residue)
+        'ResidueBase', ComponentBase)
     S = classdict['segment'] = copy_class(
-        'Segment', Segment)
-
-    # Define relationships between these classes
-    # with Level objects
-    Atomlevel = levels.Level('atom', A, AG)
-    Residuelevel = levels.Level('residue', R, RG)
-    Segmentlevel = levels.Level('segment', S, SG)
-
-    Atomlevel.parent = Residuelevel
-    Atomlevel.child = None
-    Residuelevel.parent = Segmentlevel
-    Residuelevel.child = Atomlevel
-    Segmentlevel.parent = None
-    Segmentlevel.child = Residuelevel
-
-    A.level = Atomlevel
-    AG.level = Atomlevel
-    R.level = Residuelevel
-    RG.level = Residuelevel
-    S.level = Segmentlevel
-    SG.level = Segmentlevel
+        'SegmentBase', ComponentBase)
 
     return classdict
 
 
+class _MutableBase(object):
```

Comment:
Isn't it possible to implement the new scheme without having this Mutable class object or changing the class definition in another way during runtime?